File: C/media_idades.c
#include <stdio.h>
#include <locale.h>

double media;
int idade, soma = 0, cont = 0;

int main() {
    setlocale(LC_ALL, "Portuguese");

    printf("Digite as idades: \n");
    scanf("%d", &idade);

    if (idade < 0) {
        printf("IMPOSSÍVEL CALCULAR");
    }
    else {
        while (idade > 0) {
            soma += idade;
            scanf("%d", &idade);
            cont ++;
        }

        media = (double)soma / cont;

        printf("MEDIA = %.2lf", media);
    }

}

File: Java/media_idades.java
import java.util.Locale;
import java.util.Scanner;

public class media_idades {

    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        Locale.setDefault(Locale.US);

        System.out.println("Digite as idades: ");
        int idade = sc.nextInt();

        int soma = 0;
        int cont = 0;

        if (idade < 0) {
            System.out.print("Impossível calcular");
        }
        else {
            while (idade > 0) {
                soma += idade;
                cont++;
                idade = sc.nextInt();
            }
            double media = (double)soma / cont;
            System.out.printf("Media = %.2f", media);
        }

        sc.close();

    }
}
